shared_model/validators/block_validator.hpp

```diff
@@ -37,12 +40,50 @@ namespace shared_model {
        * @return Answer containing found error if any
        */
       Answer validate(const interface::Block &block) const override {
-        return ContainerValidator<interface::Block,
-                                  FieldValidator,
-                                  TransactionsCollectionValidator>::
+        Answer answer = ContainerValidator<interface::Block,
+                                           FieldValidator,
+                                           TransactionsCollectionValidator>::
             validate(block, "Block", [this](auto &reason, const auto &cont) {
               this->field_validator_.validateHash(reason, cont.prevHash());
             });
+
+        validation::ReasonsGroupType block_reason;
+        std::unordered_set<std::string> hashes = {};
+
+        auto rejected_hashes = block.rejected_transactions_hashes();
+
+        boost::for_each(rejected_hashes | boost::adaptors::indexed(0),
+                        [&block_reason, &hashes](const auto &hash) {
+                          if (hashes.count(hash.value().hex())) {
+                            block_reason.second.emplace_back(
+                                (boost::format("Rejected hash '%s' with index "
+                                               "'%d' is not unique")
```

Comment:
I suggest being more concrete
```suggestion
                                               "'%d' has already appeared in a block")
```

Reply:
Fixed